plugin/unused-imports.vim

```diff
@@ -36,11 +36,15 @@ function! s:highlight_unused_imports(remove)
   " where does the class definition start (= where the imports end)
   call cursor(1, 1)
   let classStartLine = search('\v(^\s*import\s+)@<!(<fun>|<class>|<object>|<interface>|\@)')
-
   while linenr < classStartLine
     let linenr += 1
     let line = getline(linenr)
+    " assemble non static imports for removal
     let lis = matchlist(line, '\v^\s*import\s+(\w+\.)+(\w+);?')
+    if len(lis) == 0
+      " assemble static imports for removal
+      let lis = matchlist(line, '\v^\s*import static\s+(\w+\.)+(\w+);?')
```

Comment:
Thanks! Minor quibble here - should we check for static imports even if we find non-static imports? i.e this line will become

```
add(len, matchlist(line, '\v^\s*import static\s+(\w+\.)+(\w+);?'))
```

Reply:
sure I can do that, but doesn’t imply someone puts two imports of two different types on the same line? seems like we shouldn’t adjust for that coding style? but up to you! I can change if you prefer.